lib/simulator-xcode-8.js

```diff
@@ -94,15 +94,22 @@ class SimulatorXcode8 extends SimulatorXcode7 {
    * @override
    *
    * @param {string} bundleId - The bundle id of the application to be checked.
-   * @param {string} appFule - Application name minus ".app" (for iOS 7.1)
-   * @return {boolean} True if the given application is installed
+   * @return {boolean} True if the given application is installed.
    */
   async isAppInstalled (bundleId) {
     try {
-      let appContainer = await getAppContainer(this.udid, bundleId, false);
+      const appContainer = await getAppContainer(this.udid, bundleId, false);
       return appContainer.endsWith('.app');
     } catch (err) {
-      return false;
+      // get_app_container subcommand fails for system applications,
+      // so we try the hidden appinfo subcommand, which prints correct info for
+      // system/hidden apps
+      try {
+        const appInfo = await appInfo(this.udid, bundleId);
```

Comment:
Does this command _not_ work for regular apps? Why not use it for everything?

Reply:
it does work, but the fact that this command is hidden makes me a bit suspicious about it. The stuff might stop working completely Apple decides to change the behaviour there in some future releases. Although the current approach will still work for non-system apps, which is the most common clients case.